samza-log4j2/src/main/java/org/apache/samza/logging/log4j2/StreamAppender.java

```diff
@@ -439,7 +439,7 @@ private void startTransferThread() {
    * Helper method to send a serialized log-event to the systemProducer, and increment respective methods.
    * @param serializedLogEvent
    */
-  private void sendEventToSystemProducer(byte[] serializedLogEvent) {
+  protected void sendEventToSystemProducer(byte[] serializedLogEvent) {
```

Comment:
Another thought here is to expose the part that needs to be overridden by the extenders so that the core aspects around sending an event to `systemProducer` can remain consistent. 
It allows consistent set of metrics to be tracked and the evolution around sending events will remain closed and common.

Not sure if this is a major concern given the function as is light weight. Just throwing out another option for you to think about.

Reply:
yep, i did consider that option but decided against to avoid proliferating StreamAppender with one liner methods.